examples/with-okta-multi-tenant-pkce-flow/api-server/main.go

```diff
@@ -0,0 +1,169 @@
+						nAuthUrlGET := func(provider tpmodels.TypeProvider, options tpmodels.APIOptions, userContext supertokens.UserContext) (tpmodels.AuthorisationUrlGETResponse, error) {
+							addTenantInfoToUserContext(options.Req, userContext)
+							return oAuthUrlGET(provider, options, userContext)
+						}
+						*originalImplementation.AuthorisationUrlGET = nAuthUrlGET
+
+						oSignInUpPOST := *originalImplementation.SignInUpPOST
+						nSignInUpPOST := func(provider tpmodels.TypeProvider, code string, authCodeResponse interface{}, redirectURI string, options tpmodels.APIOptions, userContext supertokens.UserContext) (tpmodels.SignInUpPOSTResponse, error) {
+							addTenantInfoToUserContext(options.Req, userContext)
+							return oSignInUpPOST(provider, code, authCodeResponse, redirectURI, options, userContext)
+						}
+						*originalImplementation.SignInUpPOST = nSignInUpPOST
+						return originalImplementation
+					},
+
+					Functions: func(originalImplementation tpmodels.RecipeInterface) tpmodels.RecipeInterface {
+						oSignInUp := *originalImplementation.SignInUp
+						nSignInUp := func(thirdPartyID string, thirdPartyUserID string, email tpmodels.EmailStruct, userContext supertokens.UserContext) (tpmodels.SignInUpResponse, error) {
+							tenantId := (*userContext)["tenantId"].(string)
+
+							thirdPartyUserID = thirdPartyUserID + "+" + tenantId
+							return oSignInUp(thirdPartyID, thirdPartyUserID, email, userContext)
+						}
+						*originalImplementation.SignInUp = nSignInUp
+
+						return originalImplementation
+					},
+				},
+			}),
+		},
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	router := mux.NewRouter()
+
+	router.HandleFunc("/sessioninfo", session.VerifySession(nil, sessioninfo)).Methods(http.MethodGet)
+
+	http.ListenAndServe("0.0.0.0:8000", handlers.CORS(
+		handlers.AllowedHeaders(append([]string{"Content-Type"}, supertokens.GetAllCORSHeaders()...)),
+		handlers.AllowedMethods([]string{"GET", "POST", "PUT", "HEAD", "OPTIONS"}),
+		handlers.AllowedOrigins([]string{"http://multitenant.com:3000", "http://tenant1.multitenant.com:3000", "http://tenant2.multitenant.com:3000"}),
+		handlers.AllowCredentials(),
+	)(supertokens.Middleware(router)))
+}
+
+func addTenantInfoToUserContext(req *http.Request, userContext supertokens.UserContext) {
+	var config OktaConfig
+	var oktaRedirectURI string
+	tenantId := req.URL.Query().Get("tenant")
+	oktaRedirectURI = "http://multitenant.com:3000/auth/callback/okta"
+
+	switch tenantId {
+	case "tenant1":
+		config = OktaConfig{
+			ClientID:   "0oa5y2vfqhrTPAX8Q5d7",
+			OktaDomain: "dev-8636097.okta.com",
+		}
+		break
+
+	case "tenant2":
+		config = OktaConfig{
+			ClientID:   "0oa5y4z0bpdDTPOo05d7",
+			OktaDomain: "dev-8636097.okta.com",
+		}
+		break
+	}
+
+	(*userContext)["oktaConfig"] = config
+	(*userContext)["tenantId"] = tenantId
+	(*userContext)["redirectURI"] = oktaRedirectURI
+	(*userContext)["state"] = req.URL.Query().Get("state")
```

Comment:
extract this into another function `GetStateFromRequest`

Reply:
done